lib/Dialect/TTMetal/Transforms/Passes.cpp

```diff
@@ -175,25 +172,29 @@ class TTIRToTTMetalDispatchRewriter : public OpRewritePattern<ttir::GenericOp> {
       arg.setType(ty);
     }
 
-    rewriter.setInsertionPointToStart(noc0Block);
-    auto push0 = rewriter.create<ttkernel::CBPushBackOp>(
-        op.getLoc(), noc0Block->getArgument(0));
-    push0->remove();
-    noc0Block->push_back(push0);
-    auto return0 =
-        rewriter.create<ttkernel::ReturnOp>(op.getLoc(), ValueRange());
-    return0->remove();
-    noc0Block->push_back(return0);
-
-    rewriter.setInsertionPointToStart(noc1Block);
-    auto push1 = rewriter.create<ttkernel::CBPushBackOp>(
-        op.getLoc(), noc1Block->getArgument(1));
-    push1->remove();
-    noc1Block->push_back(push1);
-    auto return1 =
-        rewriter.create<ttkernel::ReturnOp>(op.getLoc(), ValueRange());
-    return1->remove();
-    noc1Block->push_back(return1);
+    {
+      OpBuilder noc0Builder(noc0Block, noc0Block->begin());
+      auto one = noc0Builder.create<arith::ConstantOp>(
```

Comment:
No that is a different constant op.  The issue linked is for creating constant tensors.  I think we should leave this and actually support / use the `arith` dialect inside of `TTKernel` because it'll give us the ability to create and to work with integers for free.